### PR TITLE
[qoc][worker] Deduplicate checkpoint and lr methods

### PR DESCRIPTION
**Pull request by CharlieFRuan**
There were redundant `save_checkpoint()`, `load_checkpoint()`, `save_hf_model()`, `get_lr()`, and `set_lr()` between `CriticWorkerBase` and `PolicyWorkerBase`. The goal might have been that `Ref` models don't need this, but we will not call them anyway. Removed ~50 LOCs.

---

<a href="https://app.devin.ai/review/novasky-ai/skyrl/pull/1598" target="_blank">
  <picture>
    <source media="(prefers-color-scheme: dark)" srcset="https://static.devin.ai/assets/gh-open-in-devin-review-dark.svg?v=1">
    Open in Devin Review
  </picture>
</a>

### CODE REVIEW

File: skyrl/backends/skyrl_train/workers/worker.py [hunk 1]
```diff
@@ -397,6 +393,51 @@ def forward(
...
+
+    def set_lr(self, learning_rate: float) -> None:
+        """
+        Set learning rate for the optimizer.
+
+        This directly updates the optimizer's param_groups, bypassing the scheduler.
+        Useful for external learning rate schedules (e.g., from Tinker).
+        """
+        for param_group in self.optimizer.param_groups:
+            param_group["lr"] = learning_rate
```
[the remark this answers is not in the document]
Comment (CharlieFRuan) [reply skyrl/backends/skyrl_train/workers/worker.py:439]: These models will not be called anyway for the ref model. But a follow-up PR will add None for self.optimizer and self.scheduler for Ref worker
